### PR TITLE
114 Add Voters for Supplier Entities

### PR DESCRIPTION
**Pull request by uplyft**

### CODE REVIEW

File: src/Controller/SupplierController.php [hunk 1]
```diff
@@ -124,8 +125,7 @@ public function update(Request $request, $id)
         /** @var Supplier $supplier */
         $supplier = $this->getSupplier($id);
 
-        // TODO: get permissions working (#1)
-        // $this->checkEditPermissions($supplier);
+        $this->denyAccessUnlessGranted('EDIT', $supplier);
```
Comment (akoebbe) [inline src/Controller/SupplierController.php:128]: Same here

File: src/Controller/SupplierController.php [hunk 2]
```diff
@@ -147,8 +147,7 @@ public function destroy($id)
     {
         $supplier = $this->getSupplier($id);
 
-        // TODO: get permissions working (#1)
-        // $this->checkEditPermissions($supplier);
+        $this->denyAccessUnlessGranted('EDIT', $supplier);
```
Comment (akoebbe) [inline src/Controller/SupplierController.php:150]: Same here

File: src/Security/SupplierVoter.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+use App\Entity\Supplier;
+use App\Entity\User;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+
+class SupplierVoter extends Voter
+{
+    const EDIT = 'EDIT';
+    const VIEW = 'VIEW';
```
Comment (akoebbe) [inline src/Security/SupplierVoter.php:12]: These should be `public const`. Something we still need to clear up in other places.

Comment (uplyft) [reply src/Security/SupplierVoter.php]: Cool, thanks.

File: src/Controller/SupplierController.php (hunk 2 above)
Comment (uplyft) [reply src/Controller/SupplierController.php]: Got it

File: src/Controller/SupplierController.php (hunk 1 above)
Comment (uplyft) [reply src/Controller/SupplierController.php]: ✓

File: src/Controller/SupplierController.php [hunk 4]
```diff
@@ -98,8 +100,7 @@ public function store(Request $request, ValidatorInterface $validator)
         $params = $this->getParams($request);
         $supplier->applyChangesFromArray($params);
 
-        // TODO: get permissions working (#1)
-        // $this->checkEditPermissions($supplier);
+        $this->denyAccessUnlessGranted('EDIT', $supplier);
```
[the remark this answers is not in the document]
Comment (uplyft) [reply src/Controller/SupplierController.php]: ✓

File: src/Controller/SupplierController.php [hunk 5]
```diff
@@ -80,6 +80,8 @@ public function show(Request $request, $id)
     {
         $supplier = $this->getSupplier($id);
 
+        $this->denyAccessUnlessGranted('VIEW', $supplier);
```
[the remark this answers is not in the document]
Comment (uplyft) [reply src/Controller/SupplierController.php]: Did you mean `SupplierVoter::VIEW` ?

**Comment by uplyft** [comment]
@akoebbe Do we want a user that doesn't have view permissions to be able to see the list of Suppliers?

**Pull request merged**